pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -131,33 +131,54 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 }
 
 func setProductLabels(cr *api.KieApp, env *api.Environment) {
-	setObjectLabels(cr, &env.Console, "business-central")
+	envConstants, _ := constants.EnvironmentConstants[cr.Status.Applied.Environment] // is this the only way to read the Image name/s ?
+	setObjectLabels(cr, &env.Console, "business-central", constants.BcImageName, getSubComponentTypeByImageName(constants.BcImageName))
 	for index := range env.Servers {
-		setObjectLabels(cr, &env.Servers[index], "kie-server")
+		/*
+		 handle the link with name->type with getSubComponentTypeByImageName
+		which is the correct field to use in the Servers to read the component ?
+		*/
+		//this will be replaced
+		setObjectLabels(cr, &env.Servers[index], "kie-server", envConstants.App.ImageName, getSubComponentTypeByImageName(envConstants.App.ImageName))
 	}
-	setObjectLabels(cr, &env.SmartRouter, "smart-router")
-	setObjectLabels(cr, &env.ProcessMigration, "process-migration")
+	setObjectLabels(cr, &env.SmartRouter, "smart-router", constants.SmartRouterImageName, getSubComponentTypeByImageName(constants.SmartRouterImageName))
+
+	//this is supposed to stay here as previous but with type application
+	setObjectLabels(cr, &env.ProcessMigration, "process-migration", constants.ProcessMigrationImageName, getSubComponentTypeByImageName(constants.ProcessMigrationImageName))
 }
 
-func setObjectLabels(cr *api.KieApp, object *api.CustomObject, component string) {
+func getSubComponentTypeByImageName(imageName string) string {
+	retValue := ""
+	switch imageName{
+		case constants.KieServerImageName, constants.BcImageName, constants.BcMonitoringImageName, constants.OperatorImageName, constants.ProcessMigrationImageName:
+			retValue = "application"
+		case constants.SmartRouterImageName, constants.ControllerImageName:
+			retValue = "infrastructure"
+	}
+	return retValue
+}
+
+
+func setObjectLabels(cr *api.KieApp, object *api.CustomObject, component string, subcomponent string, subcomponentType string) {
 	for index, obj := range object.DeploymentConfigs {
-		object.DeploymentConfigs[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, component)
+		object.DeploymentConfigs[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, component, subcomponent, subcomponentType)
```

Comment:
Would something like this be a more accurate approach?
```golang
		subcomponent, _, _ := GetImage(object.DeploymentConfigs[index].Spec.Template.Spec.Containers[0].Image)
		object.DeploymentConfigs[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, component, subcomponent, getSubComponentTypeByImageName(subcomponent))
```

Reply:
Sounds good, but how to retrieve the components who replace the single component, previously kie-server ?